src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -85,5 +169,149 @@ public override bool Execute()
 
             return true;
         }
+
+        private void DiscoverPackagesFromAllSourceBuildSources(XElement pkgSourcesElement, Hashtable allSourcesPackages, Hashtable currentPackages, Hashtable referencePackages, Hashtable previouslyBuiltPackages)
+        {
+            foreach (string packageSource in SourceBuildSources)
+            {
+                XElement sourceElement = pkgSourcesElement.Descendants().FirstOrDefault(e => e.Name == "add" && e.Attribute("key").Value == packageSource);
+                if (sourceElement == null)
+                {
+                    continue;
+                }
+
+                string path = sourceElement.Attribute("value").Value;
+                if (!Directory.Exists(path))
+                {
+                    continue;
+                }
+
+                string[] packages = Directory.GetFiles(path, "*.nupkg", SearchOption.AllDirectories);
+                foreach (string package in packages)
+                {
+                    NupkgInfo info = GetNupkgInfo(package);
+                    string id = info.Id.ToLower();
+                    string version = info.Version.ToLower();
+
+                    // Add package with version to appropriate hashtable
+                    if (packageSource.StartsWith("source-built-"))
+                    {
+                        AddToHashtable(currentPackages, id, version);
+                    }
+                    else if (packageSource.Equals("reference-packages"))
+                    {
+                        AddToHashtable(referencePackages, id, version);
+                    }
+                    else // previously built packages
+                    {
+                        AddToHashtable(previouslyBuiltPackages, id, version);
+                    }
+
+                    AddToHashtable(allSourcesPackages, packageSource, id);
+                }
+            }
+        }
+
+        private void DiscoverPackagesFromSBRPCacheSource(XElement pkgSourcesElement, Hashtable allSourcesPackages, Hashtable currentPackages)
+        {
+            // source-build-reference-package-cache is a dynamic source, populated by SBRP build,
+            // Discover all SBRP packages from checked in nuspec files.
+
+            if (string.IsNullOrEmpty(VmrRoot))
+            {
+                throw new InvalidDataException(string.Format(CultureInfo.CurrentCulture, "VmrRoot is not set - cannot determine SBRP packages."));
+            }
```

Comment:
I would probably just add the `Required` attribute to the VmrRoot property and remove this block.

Reply:
Fixed with https://github.com/dotnet/installer/pull/19114/commits/3f02bc95f3815f096425785f6fd3e2a351888a7e